### PR TITLE
Spring Security Setup

### LINKED ISSUE #13: T15.1:Setup Spring Security

### LINKED ISSUE #14: T15.2: Develop a PUT endpoint that logs a registered user into the platform

### LINKED ISSUE #16: T14.1: Setup Spring Security

### LINKED ISSUE #17: T14.2: Develop a POST endpoint in order to register a user

### PR DESCRIPTION
**Pull request by solodezaldivar**
- Add Login and Registration functionality with JWT

Closes #13
Closes #14
Closes #16
Closes #17

### CODE REVIEW

File: src/main/java/com/agileavengers/icuconnectbackend/repository/UserRepository.java [hunk 1]
```diff
@@ -1,7 +1,11 @@
 package com.agileavengers.icuconnectbackend.repository;
 
-import com.agileavengers.icuconnectbackend.model.User;
 import org.springframework.data.jpa.repository.JpaRepository;
 
+import com.agileavengers.icuconnectbackend.model.User;
+
 public interface UserRepository extends JpaRepository<User, Long> {
+    User findByUsername(String username);
```
Comment (msalze) [inline src/main/java/com/agileavengers/icuconnectbackend/repository/UserRepository.java:8]: Maybe you could use Optional<User> to ensure that if it is empty, it cannot cause a Null Pointer?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,7 +1,10 @@
+jwt.secret=HRlELXqpSB
 spring.config.import=file:env.properties
 spring.jpa.hibernate.ddl-auto=update
 spring.datasource.url=jdbc:mysql://localhost:3307/icu-connect?createDatabaseIfNotExist=true&autoReconnect=true
 spring.datasource.username=${DB_USER}
 spring.datasource.password=${DB_PASSWORD}
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.jpa.database-platform=org.hibernate.dialect.MySQLDialect
+spring.main.allow-circular-references=true
```
Comment (msalze) [inline src/main/resources/application.properties:9]: Why is this needed?

File: src/main/java/com/agileavengers/icuconnectbackend/repository/UserRepository.java (hunk 1 above)
Comment (solodezaldivar) [reply src/main/java/com/agileavengers/icuconnectbackend/repository/UserRepository.java]: Noted, made the change

File: src/main/resources/application.properties (hunk 2 above)
Comment (solodezaldivar) [reply src/main/resources/application.properties:9]: There is a circular dependency that I was not able to solve, so this is a short time fix

**Review by msalze — APPROVED** [review]
